cypress/integration/24-authentication/user-logs-in/user_logs_in.js

```diff
@@ -0,0 +1,119 @@
+import { Given, When, Then, And } from 'cypress-cucumber-preprocessor/steps';
+import { USER_CONNECT_ORCID, USER_NO_ARP, USER_NO_NAME_IN_ARP, USER_WITH_AUTHOR } from '../../../support/constants';
+
+Given('that the user logs in with their Feide ID', () => {});
+
+// @217
+// Scenario: User without their Feide ID in ARP logs in
+And('they do not have their Feide ID in any ARP entry', () => {
+  cy.login(USER_NO_NAME_IN_ARP);
+});
+Then(
+  'they see a list containing "Name", "Last registration" and "Institutions" for each ARP entry matching their name',
+  () => {
+    cy.get('[data-testid=author-name-column]');
+    cy.get('[data-testid=author-last-registration-column]');
+    cy.get('[data-testid=author-organizations-column]');
+  }
+);
+And('they see a Create New Author Button', () => {
+  cy.get('[data-testid=button-create-authority]').should('be.visible');
+});
+And('they see a Support button', () => {
+  // TODO Support button not implemented
+});
+
+// @1206
+// Scenario: User with their Feide ID in ARP logs in
+And('their Feide ID is in an ARP entry', () => {
+  cy.login(USER_WITH_AUTHOR);
+});
+Then('they can see their name in the menu', () => {
+  cy.get('[data-testid=menu]').contains('Withauthor TestUser');
+});
+
+// Common steps for @384 and @219
+And('their Feide ID is added to their Author identity', () => {
+  cy.get('@authority').its('feideids').should('have.length', 1);
+});
+And('their Organization ID \\(Cristin ID) is added to their Author identity', () => {
+  cy.get('@authority').its('orgunitids').should('have.length', 1);
+});
+
+// @384
+// Scenario: User creates a new Author identity
+And('they do not have their Feide ID in any ARP entry', () => {});
+Then('they see proposed name for a new Author identity based on data from their Feide account', () => {
+  cy.login(USER_NO_NAME_IN_ARP);
+  cy.get('[data-testid=connect-author-modal]').contains('TestUser, No name in ARP');
+});
+When('they click Create Author identity button', () => {
+  cy.get('[data-testid=button-create-authority]').click({ force: true });
+  cy.window().its('store').invoke('getState').its('user').its('authority').as('authority');
+});
+Then('this new Author identity is added to ARP', () => {});
+And('they can see confirmation message that they have connected an Author identity', () => {
+  cy.get('[data-testid=connect-author-modal]').get('[data-testid=connected-authority-heading]').should('be.visible');
+});
+
+// @219
+// Scenario: User updates an Author identity
+And('they do not have their Feide ID in any ARP entry', () => {});

```

Comment:
er det sånn at hvis du ikke oppgir en ny Given, så brukes den forrige?

Reply:
Cypress bruker den første Given som den finner noe som gjør at man må være litt oppmerksom på det når man skriver features. (Det gjelder også for When, Then og And og And er bare et alias for Given, When eller Then)

Reply:
skjønner 👍